src/app/data/devicesActions.js

```diff
@@ -0,0 +1,55 @@
+import {
+  getDocs, getDoc, doc,
+  addDoc, setDoc
+} from 'firebase/firestore';
+import { devicesCollection } from '../../app/firebase/firestoreConfig';
+import { setFormPrompt } from '../redux/form/formActions';
+
+const getAllDevices = async () => {
+  const response = await getDocs(devicesCollection);
+  const documents = response.docs;
+
+  const devices = [];
+  documents.forEach(item => {
+    devices.push({
+      id: item.id,
+      ...item.data()
+    })
+  })
```

Comment:
Could we use `map` for this one? If so, please find all similar cases and fix them